cycles-ledger/src/storage.rs

```diff
@@ -1567,25 +1672,55 @@ pub async fn withdraw(
         // subtract the fee to pay for the reimburse block
         let amount_to_reimburse = amount.saturating_sub(config::FEE);
         if amount_to_reimburse.is_zero() {
-            return Err(FailedToWithdraw {
+            return Err(FailedToWithdrawFrom {
                 fee_block: None,
+                approval_refund_block: None,
                 rejection_code,
                 rejection_reason,
             });
         }
         match reimburse(from, amount_to_reimburse, now) {
             Ok(fee_block) => {
                 prune(now);
-                return Err(FailedToWithdraw {
+                if let Some(spender) = spender {
+                    if spender != from && amount > 3 * config::FEE {
+                        match reimburse_approval(
+                            from,
+                            spender,
+                            amount_with_fee.saturating_sub(3 * config::FEE), // charge FEE for every block: withdraw attempt, refund, refund approval
```

Comment:
`amount > 2 * FEE` is indeed correct, thanks. I changed the reimburse amount to `amount_to_reimburse.saturating_sub(config::FEE)` since that feels even more clear to me how we arrive at that value